cime_config/acme/machines/cesm_prerun

```diff
@@ -1,128 +0,0 @@
-#!/bin/csh -f
-# cesm_prerun without performance data archiving: 
-#  ./cesm_prerun with the following argument keyword/value pairs, in any order
-#   -machine <system name>
-#   -compiler <compiler name> 
-#   -case <case name>
-#   -jobid <should be same as $PBS_JOBID for PBS and $COBALT_JOBID> for Cobalt>
-#   -cesmid <$LID in cesm-lingo>
-#   -user <should be same as $LOGNAME>
-#   -caseroot <directory from which job script was submitted>
-#   -rundir <run directory>
-#   -save <TRUE|true|anything else>
-#   -archive <performance archive root directory>
-#   -sampling <sampling interval in seconds>
-
-set machine = 'unknown'
-set compiler = 'unknown'
-set case = 'unknown'
-set app_jobid = 'unknown'
-set lid = 'unknown'
-set user = 'unknown'
-set caseroot = '.'
-set run_dir = '.'
-set save_timing = 'FALSE'
-set save_timing_dir = '/UNSET'
-set sample_interval = 0
-
-set i = 1
-while ($i < $#argv)
-  if      ("X$argv[$i]" == 'X-machine') then
-    @ i = $i + 1
-    set machine = $argv[$i]
-  else if ("X$argv[$i]" == 'X-compiler') then
-    @ i = $i + 1
-    set compiler = $argv[$i]
-  else if ("X$argv[$i]" == 'X-case') then
-    @ i = $i + 1
-    set case = $argv[$i]
-  else if ("X$argv[$i]" == 'X-jobid') then
-    @ i = $i + 1
-    set app_jobid = $argv[$i]
-  else if ("X$argv[$i]" == 'X-cesmid') then
-    @ i = $i + 1
-    set lid = $argv[$i]
-  else if ("X$argv[$i]" == 'X-user') then
-    @ i = $i + 1
-    set user = $argv[$i]
-  else if ("X$argv[$i]" == 'X-caseroot') then
-    @ i = $i + 1
-    set caseroot = $argv[$i]
-  else if ("X$argv[$i]" == 'X-rundir') then
-    @ i = $i + 1
-    set run_dir = $argv[$i]
-  else if ("X$argv[$i]" == 'X-save') then
-    @ i = $i + 1
-    set save_timing = $argv[$i]
-  else if ("X$argv[$i]" == 'X-archive') then
-    @ i = $i + 1
-    set save_timing_dir = $argv[$i]
-  else if ("X$argv[$i]" == 'X-sampling') then
-    @ i = $i + 1
-    set sample_interval = $argv[$i]
-  endif
-  @ i = $i + 1
-end
-
-if (-e $run_dir/cesm_prerun_done) then
-  /bin/rm $run_dir/cesm_prerun_done
-endif
-
-if ($app_jobid == 'unknown') then
-  if ($machine == 'titan') then
-
-    set app_jobid = $PBS_JOBID
-
-  else if (($machine == 'edison') || ($machine == 'corip1')) then
-
-    set app_jobid = $SLURM_JOB_ID
-
-  else if ($machine == 'mira') then
-
-    set app_jobid = $COBALT_JOBID
-
-  endif  
-endif
-
-if ($machine == 'titan') then
```

Comment:
It will break any postprocessing scripts that look for this naming 
convention. I have a few, and so do some others. I also do not like 
relying on provenance by directory name, so the files whose names I've 
taken responsibility for I've tended to append $lid to. There are also 
some files that also live outside of the performance_archive, and 
require this to prevent being overwritten. I don't think that these are 
among the ones you are touching, but would have to check to be sure.

If possible, could we start by duplicating the current design? I use 
this stuff every day, and everything in there was put there for some 
reason (through perhaps not all is currently relevant).

The current design was always meant to be a prototype - just waiting for 
you to do this correctly :-) - but it is also code that is likely to 
continue to evolve (especially when workflow starts hijacking or 
reimplementing some of this). The current design was to make it easy to 
require for ACME internal use, but not to subject others (CESM partners, 
potential external users of ACME after release) to it since they are 
unlikely to care or use it. Making it easy to further develop it without 
affecting non-ACME users is probably likely to continue to be important. 
(I feel the same way about the timing library, fyi.)

On 7/14/16 6:29 PM, James Foucar wrote:
> In cime_config/acme/machines/cesm_prerun
> <https://github.com/ESMCI/cime/pull/259#discussion_r70895626>:
>
>> -  if ($machine == 'titan') then
>> -
>> -    set app_jobid = $PBS_JOBID
>> -
>> -  else if (($machine == 'edison') || ($machine == 'corip1')) then
>> -
>> -    set app_jobid = $SLURM_JOB_ID
>> -
>> -  else if ($machine == 'mira') then
>> -
>> -    set app_jobid = $COBALT_JOBID
>> -
>> -  endif
>> -endif
>> -
>> -if ($machine == 'titan') then
>
> Will the lack of $lid suffix break anything? If not, we can have that
> discussion anytime.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ESMCI/cime/pull/259/files/302c66c97f86a90bf4ac5a730734e7d84b0a9629#r70895626>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHghFIBkSEQDjRpFwfBtyaoknhRx3LF9ks5qVrhkgaJpZM4JM3fF>.
>